### ISSUE TITLE
Fork CKAN core for the  solr-down-200-code fix 

### ISSUE THREAD

**Issue by FuhuXia** [GSA/data.gov#4704]
This following bug in CKAN core has been a pain for us given our solr periodically restarts a few times a week.  
- https://github.com/ckan/ckan/issues/8002

Now it is fixed and merged into CKAN master branch, we want to cherry-pick this change now.
- https://github.com/ckan/ckan/pull/8094

## Sketch

- Cherry-pick this [change](https://github.com/ckan/ckan/pull/8094/commits) into `ckan-2.10.4`
- Verify the fix
- Create a `ckan-2-10-4-fix` branch on GSA CKAN
- point ckan package in the catalog repo to the forked branch in the requirement.in

**Comment by FuhuXia** [comment]
The change will be included in CKAN 2.10.5. Waiting for it.

**Comment by gujral-rei** [comment]
@FuhuXia, should we move this from blocked to in progress, or are we still blocked on CKAN for adding a fix to our fork?

**Comment by FuhuXia** [comment]
Let us fix it using our fork. Don't want to wait any longer.

**Comment by FuhuXia** [comment]
Catalog and inventory are on patched CKAN core. No more need to clear cloudfront cache on solr restart.

**Comment by FuhuXia** [comment]
obviously we missed some edge cases that solr down still returns 200 code and it is cached by cloudfront. 

reopen.

**Comment by FuhuXia** [comment]
Fix deployed and verified. Getting 500 and 502 code during SOLR restarting.
